### PR TITLE
Configure flake8 and yapf with pre-commit

### LINKED ISSUE #130: Add a pre-linting file
I think there is a need to add a file that can check for linting errors before pushing changes to the remote repo.

### PR DESCRIPTION
**Pull request by vinaypuppal**
**Checklist**

- [x] My branch is up-to-date with the upstream `develop` branch.
- [x] I have added necessary documentation (if appropriate).

**Which issue does this PR fix?**: 
fixes #130 

**Why do we need this PR?**:
 To ensure that all the code we write follows the standard convention.

**What is the new behavior?**
- [x] Configured [Flake8](https://medium.com/python-pandemonium/what-is-flake8-and-why-we-should-use-it-b89bd78073f2) linter.
- [x] Configured [yapf](https://github.com/google/yapf) (a python code formatter from google).
- [x] Configured [pre-commit](http://pre-commit.com/) so `flake8` and `yapf` are run for every `git commit`.
- [x] Updated `Readme` on how to install `pre-commit` hooks.
- [x] Updated `travis.yml` to run all precommit-hooks on changed files

**Testing instructions:**
1. Change some file/s in codebase.
1. Add file/s to staging.
1. Commit file/s to git.
1. If there is some linting or formatting issue then committing will be stopped.
1. If its formatting issue `yapf` will make changes in place so add changed files to staging again and run git commit agin.
	screen shot 2017-12-25 at 22 35 12
	screen shot 2017-12-25 at 22 35 39
1. If its lint issue fix them, add changed files to staging again and run git commit agin.

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+exclude: '^$'
+fail_fast: true
+repos:
+-   repo: https://github.com/pre-commit/mirrors-yapf
```
Comment (tapaswenipathak) [inline .pre-commit-config.yaml:4]: This does not seem to be v. actively used and maintained, what do you think about enabling without mirror?

File: README.md [hunk 2]
```diff
@@ -9,48 +9,61 @@ We are trying to create a medium where people who want to start with contributin
...
+   ```
+
+1. Collectstatic files using
+
+   ```bash
+   python manage.py collectstatic
+   ```
+
+1. Run the app
+   ```bash
```
Comment (tapaswenipathak) [inline README.md:64]: thanks for this cleanup! 🎉  I was planning to do this from quite a while but didn't.

File: .flake8 [hunk 3]
```diff
@@ -0,0 +1,6 @@
+[flake8]
+ignore =
+    # A continuation line is under-indented for a visual indentation.
+    E128
+
+max-line-length=80
```
Comment (tapaswenipathak) [inline .flake8:6]: ..and moving it to separate dotfile is perfect!

**Review by tapaswenipathak — COMMENTED** [review]
Thanks for suggesting and than working on this, left a comment otherwise it is 🚀

File: .pre-commit-config.yaml (hunk 1 above)
Comment (vinaypuppal) [reply .pre-commit-config.yaml:4]: > This does not seem to be v. actively used and maintained

I am curious how did you decide this?

I see `https://github.com/pre-commit/mirrors-yapf` is updated whenever `yapf`(https://github.com/google/yapf/releases) is updated as you can see here https://github.com/pre-commit/mirrors-yapf/releases 

> what do you think about enabling without mirror
- We need to provide git repository to enable pre-commit hooks (Ref: http://pre-commit.com/#plugins)
- So What you mean by without mirror?

Comment (tapaswenipathak) [reply .pre-commit-config.yaml:4]: > > This does not seem to be v. actively used and maintained

> I am curious how did you decide this?

> I see https://github.com/pre-commit/mirrors-yapf is updated whenever yapf(https://github.com/google/yapf/releases) is updated as you can see here https://github.com/pre-commit/mirrors-yapf/releases

Thanks for linking this, my observation was 👎 skipped over `53 releases`. 👍 

> > what do you think about enabling without mirror

> We need to provide git repository to enable pre-commit hooks (Ref: http://pre-commit.com/#plugins)
o w/ the number of releases they had, don't need this.

> So What you mean by without mirror?

☝️ what you understood, a fork of our own or code to enable a pre-commit hook.

File: README.md [hunk 4]
```diff
@@ -9,48 +9,61 @@ We are trying to create a medium where people who want to start with contributin
...
+   git clone git@github.com:OpenSourceHelpCommunity/OpenSourceHelpCommunity.github.io.git
+   ```
+
+1. Go inside main Django app [Instructional video on installing Django](https://youtu.be/qgGIqRFvFFk)
+
+   ```bash
+   cd oshc
+   ```
+
+1. Install [pre-commit](pre-commit.com) into your git hooks. [pre-commit](pre-commit.com) will now run on every commit. Every time you clone a project that is using [pre-commit](pre-commit.com) running [pre-commit](pre-commit.com) install should always be the first thing you do after installing requirements.
```
Comment (tapaswenipathak) [inline README.md:51]: in markdown it takes relative path if not put w/ the protocol, need a correction here.

Comment (vinaypuppal) [reply README.md:51]: oh, I missed protocol in URL somehow during copy-paste or maybe my editor removed them.

It seems you fixed it after merging, thanks.

**Pull request merged**